App/src/Service/GroupExpenseBalancer.php

```diff
@@ -41,35 +51,67 @@ private function setExpenses($expenses, $bilans)
         }
     }
 
-    private function updateBilan($bilans, $amount, $participants, $payer, $amountByParticipants)
+    /**
+     * @param Collection<User> $participants
+     * @param array<Bilan> $bilans
+     */
+    private function updateBilan(array $bilans, int $amount, Collection $participants, User $payer, int $amountByParticipants): void
     {
         foreach ($bilans as $bilan) {
+            $payerName = $payer->getName();
             $name = $bilan->getName();
             $participation = $bilan->getParticipation();
             $cost = $bilan->getCost();
             $owe = $bilan->getOwe();
 
-            if ($payer === $name) {
+            if ($payerName === $name) {
                 $bilan->setCost($cost + $amount);
             }
 
-            if (in_array($name, $participants)) {
+            if ($participants->exists(static fn ($key, User $user) => $user->getName() === $name)) {
                 $bilan->setParticipation($participation + $amountByParticipants);
             }
 
-            foreach ($participants as $participant) {
-                if ($name !== $payer || $name === $participant) {
-                    continue;
-                }
+            $this->updateParticipantOwe($participants, $name, $payerName, $owe, $amountByParticipants, $bilan, $cost, $participation);
+        }
+    }
+
+    private function updateParticipantOwe(Collection $participants, string $name, string $payerName, array $owe, int $amountByParticipants, Bilan $bilan, int $cost, int $participation): void
+    {
+        foreach ($participants as $participant) {
+            $participantName = $participant->getName();
+            if ($name !== $payerName || $name === $participantName) {
+                continue;
+            }
 
-                if (array_key_exists($participant, $owe)) {
-                    $owe[$participant] += $amountByParticipants;
-                }
+            if (array_key_exists($participantName, $owe)) {
+                $owe[$participantName] += $amountByParticipants;
+            }
+
+            $owe[$participantName] = $amountByParticipants;
+        }
+
+        $bilan->setBalance($cost - $participation);
+        $bilan->setOwe($owe);
+    }
+
+    public function showBalance(int $id): array
+    {
+        $expenses = $this->expenseRepository->getExpensesOfGroupById($id);
+
+        $bilans = $this->expenseBalancer($expenses);
+
+        $balances = [];
+
+        foreach ($bilans as $bilan) {
+            $name = $bilan->getName();
+            $owe = $bilan->getOwe();
 
-                $owe[$participant] = $amountByParticipants;
+            foreach ($owe as $key => $values) {
+                $formatedValue = $values / 100;
+                array_push($balances, ["userOwe" => $key, "amount" => $formatedValue, "to" => $name]);
```

Comment:
Pareil pour tes déclarations de tableau : n'hésite pas à sauter des lignes quand ils contiennent plusieurs éléments, surtout quand ils sont associatifs, pour gagner en lisibilité 👍 

```suggestion
                $balances[] = [
                    "userOwe" => $key,
                    "amount" => $formatedValue,
                    "to" => $name,
                ];
```

Reply:
done